services/provider/server/server.go

```diff
@@ -687,6 +694,15 @@ func (s *OCSProviderServer) GetStorageClaimConfig(ctx context.Context, req *pb.S
 				"csi.storage.k8s.io/controller-expand-secret-name": provisionerSecretName,
 			}
 
+			storageClusters, err := util.GetStorageClustersInNamespace(ctx, s.client, s.namespace)
+			if err != nil || len(storageClusters.Items) == 0 {
+				return nil, status.Errorf(codes.Internal, "failed to get storage cluster %v", err)
+			}
+			kernelMountOptions := util.GetCephFSKernelMountOptions(&storageClusters.Items[0])
```

Comment:
how are you getting `ms_mode=secure` as mentioned in the test comment as the function will return `ms_mode=legacy` if remote consumers are allowed? Is it due to `sc.Spec.Network.Connections.Encryption.Enabled`?

Reply:
Yes if `sc.Spec.Network.Connections.Encryption.Enabled` then it will return ms_mode=secure